src/libraries/System.Console/src/System/ConsolePal.Wasi.cs

```diff
@@ -18,24 +18,20 @@ namespace System
     //       to also change the test class.
     internal static partial class ConsolePal
     {
-        // StdInReader is only used when input isn't redirected and we're working
-        // with an interactive terminal.  In that case, performance isn't critical
-        // and we can use a smaller buffer to minimize working set.
-        // there is no dup on WASI
         public static Stream OpenStandardInput()
         {
-            return new UnixConsoleStream(Interop.CheckIo(Interop.Sys.FileDescriptors.STDIN_FILENO), FileAccess.Read,
+            return new UnixConsoleStream(Interop.Sys.FileDescriptors.STDIN_FILENO, FileAccess.Read,
```

Comment:
I am not familiar with this particular overload, let me share it here think loud about that:

https://github.com/dotnet/runtime/blob/487d7f010b90a82178d55eea21e72dc6dedf9d5d/src/libraries/Common/src/Interop/Unix/Interop.IOErrors.cs#L86-L105

1. The xml comment does not describe what it does.
2. If provided handle is invalid, then it gets last error for the current thread and throws an exception. 

I don't think that we need such check here, as we have not performed any IO yet (the handle is created by providing a constant number: 0, 1 or 2, it's not a result of a sys-call), so there is no last error that makes sense. It can become invalid once we try to use it in a sys-call, but other layers are ready for that.

So I am fine with removing these checks (and the method itself if it's not used elsewhere) 👍